src/misc.c

```diff
@@ -162,15 +162,13 @@ char *fgetll(char *s, size_t size, FILE *stream)
 			ungetc(c, stream);
 		return s;
 	}
-	cp = s;
+	cp = strdup(s);
+
 	while (1) {
-		cp2 = cp;
-		cp = mem_alloc(len+60000);
-		strcpy(cp, cp2);
-		if (cp2 != s)
-			MEM_FREE(cp2);
+		cp = realloc(cp, 2 * len);
+
```

Comment:
Yea.  The original code itself was just a hack. It 'works', but was not optimal. For the initial size I was working with, it was fine. But for having to deal with data in the 100k to multiple GB in size, it was not adequate.